pkg/operator/mds/controller.go

```diff
@@ -84,13 +84,18 @@ func (c *FilesystemController) onAdd(obj interface{}) {
 }
 
 func (c *FilesystemController) onUpdate(oldObj, newObj interface{}) {
-	//oldFilesystem := oldObj.(*Filesystem)
-	newFilesystem := newObj.(*Filesystem)
+	oldFS := oldObj.(*Filesystem)
+	newFS := newObj.(*Filesystem)
+	if !filesystemChanged(oldFS.Spec, newFS.Spec) {
+		logger.Debugf("filesystem %s not updated", newFS.Name)
+		return
+	}
 
 	// if the file system is modified, allow the file system to be created if it wasn't already
-	err := newFilesystem.Create(c.context, c.versionTag, c.hostNetwork)
+	logger.Infof("updating filesystem %s", newFS)
+	err := newFS.Create(c.context, c.versionTag, c.hostNetwork)
```

Comment:
will recreating the filesystem update the filesystem with the new changes?

Reply:
it is intended to be idempotent. If the pools and file system already exist, it will just return success. If there are changed properties (different replication setting), it will be changed.